pkg/asset/imagebased/image/imagebased_config.go

```diff
@@ -0,0 +1,409 @@
+package image
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	dockerref "github.com/containers/image/v5/docker/reference"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent/manifests/staticnetworkconfig"
+	"github.com/openshift/installer/pkg/types/imagebased"
+	"github.com/openshift/installer/pkg/validate"
+)
+
+var (
+	configFilename = "imagebased-installation-config.yaml"
+
+	defaultExtraPartitionLabel  = "varlibcontainers"
+	defaultExtraPartitionStart  = "-40G"
+	defaultExtraPartitionNumber = uint(5)
+)
+
+// ImageBasedInstallationConfig reads the imagebased-installation-config.yaml file.
+type ImageBasedInstallationConfig struct { // nolint:revive // although this name stutters it is useful to convey that it's an image-based installer related struct
+	File     *asset.File
+	Config   *imagebased.InstallationConfig
+	Template string
+}
+
+var _ asset.WritableAsset = (*ImageBasedInstallationConfig)(nil)
+
+// Name returns a human friendly name for the asset.
+func (*ImageBasedInstallationConfig) Name() string {
+	return "Image-based Installation ISO Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*ImageBasedInstallationConfig) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the Image-based Installation Config YAML manifest.
+func (i *ImageBasedInstallationConfig) Generate(dependencies asset.Parents) error {
+	configTemplate := `#
+# Note: This is a sample ImageBasedInstallationConfig file showing
+# which fields are available to aid you in creating your
+# own imagebased-installation-config.yaml file.
+#
+apiVersion: v1beta1
+kind: ImageBasedInstallationConfig
+metadata:
+  name: example-imagebased-installation-config
+seedImage: quay.io/openshift-kni/seed-image:4.16.0
+seedVersion: 4.16.0
+installationDisk: /dev/vda
+pullSecret: '<your-pull-secret>'
+networkConfig:
```

Comment:
I wonder if we should have network configuration in template, i mean should not we set only required fields in template?

Reply:
Maybe we can set those fields as commented out ? so user can see them but not as default?